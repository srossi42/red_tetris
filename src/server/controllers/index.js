module.exports = {
    newPlayer: require('./newPlayer'),
    startGame: require('./startGame'),
    disconnectRoom: require('./disconnectRoom'),
    newShapes: require('./newShapes'),
    score: require('./score'),
    level: require('./level'),
    endGame: require('./endGame'),
    restart: require('./restart'),
  };