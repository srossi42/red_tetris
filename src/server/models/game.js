class Game {
    constructor(start, room) {
        this.start = start
        this.room = room
    }

    get start() {
        return this.start
    }
}

export default Game