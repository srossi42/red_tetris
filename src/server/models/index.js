export { Game } from './game'
export { Tetriminos } from './tetriminos'
export { Player } from './player'