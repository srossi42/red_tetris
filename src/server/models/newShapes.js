import { shapesRandom } from './shapes'

const newShapes = (random) => {
  const copyObject = shapesRandom
  return JSON.parse(JSON.stringify(copyObject[random]))
}

export default newShapes