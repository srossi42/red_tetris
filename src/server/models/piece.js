const shapes = [
    {
        shapes: [
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,1,1,-1,1,0,0,0]
        ],
        color: '#FF5733'
    },
    {
        shapes: [
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,1,0,0,0,0,0,0],
            [0,0,0,1,-1,1,0,0,0,0]
        ],
        color: '#FFBB33'
    },
    {
        shapes: [
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,0,1,0,0,0,0],
            [0,0,0,1,-1,1,0,0,0,0]
        ],
        color: '#D7FF33'
    },
    {
        shapes: [
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,1,1,0,0,0,0],
            [0,0,0,0,1,1,0,0,0,0]
        ],
        color: '#33FFFF'
    },
    {
        shapes: [
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,1,1,0,0,0,0],
            [0,0,0,1,-1,0,0,0,0,0]
        ],
        color: '#FF3371'
    },
    {
        shapes: [
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,1,0,0,0,0,0],
            [0,0,0,1,-1,1,0,0,0,0]
        ],
        color: '#9C33FF'
    },
    {
        shapes: [
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,0,0,0,0,0,0,0],
            [0,0,0,1,1,0,0,0,0,0],
            [0,0,0,0,-1,1,0,0,0,0]
        ],
        color: '#FF33C7'
    },
]

class Piece {
    constructor() {
        this.shapes = [shapes[Math.floor(Math.random() * 7)]]
    }

    shapes(tour) {
        return this.shapes[tour]
    }
}

export default Piece
