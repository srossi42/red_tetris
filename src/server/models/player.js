export class Player {
    constructor(name, socketId) {
        this.start = true
        this.name = name
        this.tour = 0
        this.socketId = socketId
    }

    tour () {
        this.tour = this.tour + 1
    }
}