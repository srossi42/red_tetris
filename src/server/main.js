var http = require('http');
var express = require('express');
var app = express();

import { disconnectRoom, newPlayer, startGame, newShapes, score, level, endGame, board, restart } from './controllers/index'

let games = []

var server = http.createServer(app);
// Pass a http.Server instance to the listen method
var io = require('socket.io').listen(server);

// The server should start listening
server.listen(3004, (err) => {
    if (err) throw console.log(err);
    console.log(`Server listen : ${3004}`);
});

// Register the index route of your app that returns the HTML file
app.get('/', function (req, res) {
    res.sendFile(__dirname + '/bundle.js' ? '/../../build/bundle.js' : '/../../index.html');
});

// Expose the node_modules folder as static resources (to access socket.io.js in the browser)
app.use('/static', express.static('node_modules'));

// Handle connection
io.on('connection', function (socket) {
    const query = socket.handshake['query']
    if (query.room !== '') {
      const arrayQuery = query.room.split('[')
      socket.addRoom = arrayQuery[0]
      socket.name = arrayQuery[1].substr(0, arrayQuery[1].length - 1)
      console.log('user join ' + socket.id, 'avec le nom : ' + socket.name + ' dans la room ' + socket.addRoom)
    }
    socket.on('newPlayer', () => {
      games = newPlayer(socket, games)
    })
    socket.on('disconnect', () => {
      games = disconnectRoom(socket, games)
    })
    socket.on('end', () => {
      games = endGame(socket, io, games)
    })
    socket.on('action', (action) => {
      switch (action.type) {
        case 'start' :
          games = startGame(socket, io, games)
          break;
        case 'shapes' : 
          games = newShapes(socket, io, games, action.board, action.length)
          break;
        case 'score' : 
          games = score(socket, action.score, games)
          break;
        case 'level' : 
          games = level(socket, action.ope, games, io)
          break;
        case 'restart' : 
          games = restart(socket, io, games)
          break;
      }
    })
});

