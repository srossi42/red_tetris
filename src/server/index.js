import fs  from 'fs'
import debug from 'debug'
import shapes from './models/piece'
import playerJoint from './models/playerJoint'
import socketMiddleWare from '../client/middleware/socketMiddleWare'
import {game} from './models/game'


const logerror = debug('tetris:error')
  , loginfo = debug('tetris:info')

const initApp = (app, params, cb) => {
  const {host, port} = params
  const handler = (req, res) => {
    const file = req.url === '/bundle.js' ? '/../../build/bundle.js' : '/../../index.html'
    fs.readFile(__dirname + file, (err, data) => {
      if (err) {
        logerror(err)
        res.writeHead(500)
        return res.end('Error loading index.html')
      }
      res.writeHead(200)
      res.end(data)
    })
  }

  app.on('request', handler)

  app.listen({host, port}, () =>{
    loginfo(`tetris listen on ${params.url}`)
    cb()
  })
}



const initEngine = (io, games) => {
  io.on('connection', function(socket){
    loginfo("Socket connected: " + socket.id)
    socket.on('co', (action) => {
      const query = socket.handshake['query']
      if (query.room !== '') {
        if (socket.adapter.rooms[socket.room] && socket.adapter.rooms[socket.room].length > 0) {
          playerJoint(socket, query)
          socket.emit('text', {text: "Waiting to game start...", name: socket.name})
        } else {
          playerJoint(socket, query)
          socket.emit()
          socket.emit('text', {text: "You're Master, Press <Enter> for START", name: socket.name})
        }
      } else {
        socket.emit('text', {text: "You're Master, Press <Enter> for START", name: ''})
      }
    })
    // socket.on('action', (action) => {
    //   if (action.type === 'start') {
    //     socket.emit('start', {
    //       shapes: shapes(),
    //         newShapes: shapes()
    //     })
    //   } else if (action.type === 'shapes') {
    //     socket.emit('shapes', {
    //       newShapes: shapes(),
    //     })
    //   }
    // })
    // socket.on('event', (event) => {
    //   console.log(event);
    // })
  })
}

export function create(params){
  const promise = new Promise( (resolve, reject) => {
    const app = require('http').createServer()
    initApp(app, params, () =>{

      const games = []
      const io = require('socket.io')(app)
      // const stop = (cb) => {
      //   io.close()
      //   app.close( () => {
      //     app.unref()
      //   })
      //   loginfo(`Engine stopped.`)
      //   cb()
      // }

      initEngine(io, games)
      // resolve({stop})
    })
  })
  return promise
}
