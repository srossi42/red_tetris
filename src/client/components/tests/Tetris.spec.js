import React from 'react'
import { shallow, configure } from 'enzyme';
import { Provider } from 'react-redux'
import Tetris from '../Tetris';
import { createStore } from 'redux'
import reducer from '../../reducers'
import { Tetriminos } from '../../../server/models'
import toJson from "enzyme-to-json"

const initialState = {}

const store = createStore(
  reducer,
  initialState,
)

describe('<Tetris/> Component', () => {
    const object = {
        start: true,
        win: false,
        end: false,
        score: 0,
        level: 0,
        delay: 1000,
        scoreAdversary: [],
        boardAdversary: [],
        shapes: [
            {
                shape: [
                    ["I", "I", "I", "I"],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0]
                ],
                color: "80, 227, 230"
            },
            {
                shape: [
                    ["I", "I", "I", "I"],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0]
                ],
                color: "80, 227, 230"
            },
            {
                shape: [
                    ["I", "I", "I", "I"],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0]
                ],
                color: "80, 227, 230"
            },
            {
                shape: [
                    ["I", "I", "I", "I"],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0]
                ],
                color: "80, 227, 230"
            },
            {
                shape: [
                    ["I", "I", "I", "I"],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0],
                    [0, 0, 0, 0]
                ],
                color: "80, 227, 230"
            },
        ],
        text: '',
        textEnd: '',
        name: '',
        numberPlayer: 0,
        lineFull: 0
    }
    it('rendering correctly Tetris', () => {
        let wrapper = shallow(
        <Provider store={store}>
        <Tetris
        endGame={() => null}
        newShapes={() => null}
        handleScore={() => null}
        socket={() => null}
        store={object}/>
        </Provider>
        )
        console.log(wrapper.children().debug());
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('win', () => {
        object.win = true
        let wrapper = shallow(
        <Provider store={store}>
        <Tetris
        endGame={() => null}
        newShapes={() => null}
        handleScore={() => null}
        socket={() => null}
        store={object}/>
        </Provider>
        )
        expect(wrapper).toMatchSnapshot()
    })
})
