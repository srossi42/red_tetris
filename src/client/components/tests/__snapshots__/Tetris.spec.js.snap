// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tetris/> Component rendering correctly Tetris 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Tetris
    endGame={[Function]}
    handleScore={[Function]}
    newShapes={[Function]}
    socket={[Function]}
    store={
      Object {
        "boardAdversary": Array [],
        "delay": 1000,
        "end": false,
        "level": 0,
        "lineFull": 0,
        "name": "",
        "numberPlayer": 0,
        "score": 0,
        "scoreAdversary": Array [],
        "shapes": Array [
          Object {
            "color": "80, 227, 230",
            "shape": Array [
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
            ],
          },
          Object {
            "color": "80, 227, 230",
            "shape": Array [
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
            ],
          },
          Object {
            "color": "80, 227, 230",
            "shape": Array [
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
            ],
          },
          Object {
            "color": "80, 227, 230",
            "shape": Array [
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
            ],
          },
          Object {
            "color": "80, 227, 230",
            "shape": Array [
              Array [
                "I",
                "I",
                "I",
                "I",
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
              Array [
                0,
                0,
                0,
                0,
              ],
            ],
          },
        ],
        "start": true,
        "text": "",
        "textEnd": "",
        "win": false,
      }
    }
  />
</ContextProvider>
`;

exports[`<Tetris/> Component win 1`] = `ShallowWrapper {}`;
